src/ledger/TrustLineWrapper.cpp

```diff
@@ -194,6 +194,16 @@ TrustLineWrapper::deactivate()
     mImpl.reset();
 }
 
+std::unique_ptr<TrustLineWrapper::AbstractImpl> const&
+TrustLineWrapper::getImpl() const
+{
+    if (!(*this))
+    {
+        throw std::runtime_error("TrustLineWrapper not active");
```

Comment:
I think this is more a std::logic_error than runtime_error

Reply:
Agreed this does fall more into the category of `std::logic_error`. The same would apply to most of the other errors that can be thrown by the new data layer actually. But right now we never use `std::logic_error` in stellar-core, although I am not sure why. I think I'll leave this for now and maybe at some point we can try to figure out which exceptions should fall into which categories.